sdk/inc/azure/core/az_result.h

```diff
@@ -141,6 +141,9 @@ typedef enum
   /// Error while parsing HTTP response header.
   AZ_ERROR_HTTP_CORRUPT_RESPONSE_HEADER = _az_RESULT_MAKE_ERROR(_az_FACILITY_HTTP, 7),
 
+  /// There are no more headers within the HTTP response payload.
+  AZ_ERROR_HTTP_NO_MORE_HEADERS = _az_RESULT_MAKE_ERROR(_az_FACILITY_HTTP, 8),
```

Comment:
I prefer AZ_ERROR_END_OF_HEADERS.
but this is a minor, don't really care too much

Reply:
I like that. @danewalton - maybe we should rename `AZ_ERROR_IOT_NO_MORE_PROPERTIES` to `AZ_ERROR_IOT_END_OF_PROPERTIES` too. What do you think.

Reply:
> I prefer AZ_ERROR_END_OF_HEADERS.

I am going to assume you meant `AZ_ERROR_HTTP_END_OF_HEADERS`, @vhvb1989

Reply:
@JeffreyRichter thoughts on the name?

Reply:
I think this is fine. If I did anything I might do AZ_ERROR_END_OF_**RESPONSE**_HEADERS. But this is long. Your call.

Reply:
@ahsonkhan yea that's probably a better name. Issuing PR

Reply:
> I think this is fine. If I did anything I might do AZ_ERROR_END_OF_RESPONSE_HEADERS. But this is long. Your call.

Will we ever have such errors for the request, i.e.:
AZ_ERROR_HTTP_END_OF_**REQUEST**_HEADERS

Reply:
It's hard to predict the future. I think it's unlikely as it would change our architecture a good bit.
But we might have other kinds of headers (not HTTP).
My suggestion for the RESPONSE part of the error is just to help diagnostics - it gives more information as opposed to less. But, I don't feel strongly on this one.

Reply:
I understand. The only thing holding me back is trying to reach a balance between providing lots of details that helps diagnostics and having too many different error codes. The more specific we get, the less they can be used for more than one API, and now we have a unique az_result for every function, and none can be re-used.

Trying to encode all the helpful/relevant diagnosing information in the name of an enum value won't scale.

Reply:
I don't think there is a need to avoid lots of error codes. az_result is 32-bits which gives us 4 billion possibilities. Even within just 1 facility, we have 16 bits which is 65,000. So, we're not in danger of running out. There are 2 purposes of error codes:
1. The customer's code expects it and has an alternate code path to take based on it. The more specific, the more likely they know exactly what's wrong and why they should take that code path.
2. At development time, the customer gets an error which they did NOT expect and they need to modify their code to avoid the error happening after the code is deployed. Again, the more specific the code, the moer insight we give them to aid them in understanding what went wrong and how to modify their code.